oauth2_provider/models.py

```diff
@@ -123,6 +129,22 @@ def clean(self):
     def get_absolute_url(self):
         return reverse('oauth2_provider:detail', args=[str(self.id)])
 
+    def get_allowed_scopes_from_scopes(self, scopes):
+        """
+        Filters the given list of scopes so it only contains the allowed scopes
+        """
+        if isinstance(scopes, six.string_types):
```

Comment:
This can be simplified:

```python
# When would scopes NOT be a string?
scopes = scopes.split(' ')
scopes = set(scopes)

allowed_scopes = self.allowed_scopes or ''
allowed_scopes = allowed_scopes.split(' ')
allowed_scopes = set(allowed_scopes)

scopes = allowed_scopes & scopes
return ' '.join(scopes)
```